### PR TITLE
Add metadata for GHC 9.8.2

### PR DESCRIPTION
**Pull request by wz1000**
https://gitlab.haskell.org/ghc/ghc/-/pipelines/90382

### CODE REVIEW

**Comment by wz1000** [comment]
https://github.com/haskell/ghcup-metadata/actions/runs/8004658243

**Comment by wz1000** [comment]
We need another backport to fix the darwin failure https://gitlab.haskell.org/ghc/ghc/-/merge_requests/12117

**Comment by wz1000** [comment]
This should be ready now:

https://gitlab.haskell.org/ghc/ghcup-ci/-/pipelines/90509

https://github.com/haskell/ghcup-metadata/actions/runs/8017727141

**Review by hasufell — APPROVED** [review]
Feel free to merge

**Comment by wz1000** [comment]
@hasufell what is the workflow for merging `develop` into `master` now?

**Comment by hasufell** [comment]
I haven't thought deeply about it. I guess you can just create PRs against master going forward since signing is done in the PR and we're not batching anything. 

For this instance just create a develop -> master PR.

**Comment by hasufell** [comment]
> Is it on purpose that this just updated the -vanilla file, but not `ghcup-0.0.x.yaml`?

Yes, `ghcup-0.0.x.yaml` is maintained by GHCup developers, not by upstream.

**Comment by hasufell** [comment]
> Aha. I don't understand this sentence in the README then:

It is outdated. Upstream maintains -vanilla, the rest is maintained by GHCup developers.

> So there seems to be no downside to just updating that file as well when introducing a new version?

There is, see https://github.com/haskellfoundation/tech-proposals/pull/61 for more information. GHCup has been cleaning up after upstream developers for a while now, fixing bindists, building for missing platforms etc.

However, we provide vanilla for users who don't care about bindist quality, but want to just fire up the latest release/bindist, even if it's utterly broken.

See also https://hasufell.github.io/posts/2023-11-14-ghcup-is-not-an-installer.html

File: ghcup-vanilla-0.0.7.yaml [hunk 1]
```diff
@@ -4040,12 +4040,132 @@ ghcupDownloads:
...
+        dlOutput: ghc-9.8.2-src.tar.xz
+        dlSubdir: ghc-9.8.2
+        dlUri: https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-src.tar.xz
+      viTags:
+      - Latest
+      - base-4.19.1.0
+      viTestDL:
+        dlHash: e1e35736205f555bceeecd3a78b9e8f0710c51a875149ebf5795eb88deb38a60
+        dlOutput: ghc-9.8.2-testsuite.tar.xz
+        dlSubdir: ghc-9.8.2
```
Comment (hasufell) [inline ghcup-vanilla-0.0.7.yaml:4167]: This is wrong btw @wz1000

File: ghcup-vanilla-0.0.7.yaml [hunk 2]
```diff
@@ -4040,12 +4040,132 @@ ghcupDownloads:
...
+            unknown_versioning:
+              dlHash: 67be089dedbe599d911efd8f82e4f9a19225761a3872be74dfd4b5a557fb8e1a
+              dlOutput: ghc-9.8.2-aarch64-darwin.tar.xz
+              dlSubdir: ghc-9.8.2-aarch64-apple-darwin
+              dlUri: https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-aarch64-apple-darwin.tar.xz
+          Linux_Alpine:
+            unknown_versioning:
+              dlHash: 0e1a8156ec82de77a7160db023b297cf00ab1e04e395f1e8b77f602ba051b1d5
+              dlOutput: ghc-9.8.2-aarch64-alpine3_18-linux.tar.xz
+              dlSubdir: ghc-9.8.2-x86_64-unknown-linux
```
Comment (hasufell) [inline ghcup-vanilla-0.0.7.yaml:4146]: This subdirectory is not correct

**Pull request merged**